### ISSUE TITLE
run highlighted part of query only not working in beta3

### ISSUE THREAD

**Issue by beRto-** [sqlitebrowser/sqlitebrowser#1708]
Details for the issue
--------------------

When highlighting only a piece of a query, I expect DB4S to run only that part of the query. This seemed to work as expected before, but not in Beta 3. Instead, Beta 3 continues to run the full query, rather than just the selected portion.

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [x] Windows: ( _version:_ 7 )
- [ ] Linux: ( _distro:_ ___ )
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [x] 3.11.0-alpha1 or 3.11.0-beta*  << Version 3.10.202 (Dec 13 2018)
- [ ] 3.10.1
- [ ] Other: ___

#### Did you also

- [x] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

EDIT - attaching screenshots and a summary on how to reproduce issue
[bug report.docx](https://github.com/sqlitebrowser/sqlitebrowser/files/2752170/bug.report.docx)

**Comment by justinclift** [comment]
Thanks @beRto-, that sounds like a regression. :frowning: 

Probably something to do with the refactoring of the Execute SQL tab we were doing a while back in https://github.com/sqlitebrowser/sqlitebrowser/pull/1575#issuecomment-430964392.

https://github.com/sqlitebrowser/sqlitebrowser/issues/1491#issuecomment-439152276 is possibly related to this too.

**Comment by chrisjlocke** [comment]
@beRto- can you confirm the steps you're doing.  This is handy for troubleshooting.
For example, what SQL statement have you inputted into the window?  Which area of this SQL are you highlighting?  Are you pressing <kbd>F5</kbd> (or pressing the 'Execute All' button) or pressing <kbd>Shift</kbd>+<kbd>F5</kbd> to execute the SQL?

**Comment by beRto-** [comment]
@chrisjlocke - thanks for looking into this. I attached a Word document with the table schema, data, step summary, and screenshots. I hope that helps, and let me know if you need anything else!

**Comment by chrisjlocke** [comment]
GitHub doesn't support attachments. Can you email me the document? Db4s [at] chrisjlocke dot co dot uk.

**Comment by justinclift** [comment]
@chrisjlocke There's a link to it in the updated initial bug report post up top. :smile:

&nbsp; &nbsp; https://github.com/sqlitebrowser/sqlitebrowser/files/2752170/bug.report.docx

The attaching in this case looks to have been done manually (eg it worked), rather than via email (which doesn't work). :wink:

**Comment by chrisjlocke** [comment]
Thanks @beRto-.  Can confirm this is broken in the current nightly.  It appears the 'Run SQL' is ignoring any selection.

**Comment by justinclift** [comment]
Damn.  Blocker bug for 3.11.0?

**Comment by chrisjlocke** [comment]
Wouldn't say its a blocker - it can be worked around.  Just copy the SQL to another tab, or add -- markers to comment out unnecessary SQL.  Priority-wise, its low compared to other issues.

**Comment by beRto-** [comment]
Really?! I use this highlight feature all the time when writing any non-trivial query. I would have to think most "power" users would? If I had to copy-paste or comment out every time, it would certainly prevent me from upgrading to the new version (which would be a shame, because it has some features I've been waiting on). :(

**Comment by chrisjlocke** [comment]
Appreciate that - just commenting based on our pool of developers and the time they have available.  With ~300 outstanding issues, some of those (in my opinion only - and I'm not technically on the team of developers - mainly troubleshooting and support) rank higher than this issue.
I'm not saying its not going to get resolved - it could be a trivial one line fix that someone could rattle off in 5 minutes - but it shouldn't stop 3.11.0 going out.  3.10.1 is way too old to be our 'current release', and the benefits/advantages of 3.11.0 (albeit buggy) outweigh the disadvantages..  again, in my opinion.

**Comment by justinclift** [comment]
@MKleusberg @mgrojo Any idea how much effort this will take to get fixed?

Kind of guessing it could be a pain, as the execute SQL parsing seems to be tricky and full of weird corner cases/gotchas.

**Comment by beRto-** [comment]
@chrisjlocke Fair enough, thanks for clarifying. I misunderstood that your reply meant a fix would definitely not be included in the release. Now I get your point that it *may* yet get fixed, but should not be classified as a showstopper.

I mostly agree with that assessment, but would have concern (outside of my previous point on it being "annoying"), that this bug essentially changes output behaviour without any warning to users. If someone runs a highlighted portion, which always worked a certain way before, DB4S will still return results - but the results will now be for some _other_ query than a user is expecting. Unless the user is aware of the bug or studies the output log window, they could easily be misled?

**Comment by MKleusberg** [comment]
Thanks, @beRto-! That's definitely a regression we need to fix :smile: 

I just pushed two commits, one in master and one in v3.11.x, to fix this issue. Because the code has changed a lot between these versions we might want to test both branches though.

@beRto- Can you download tomorrow's nightly build and see if it's working as expected for you using that?

**Comment by beRto-** [comment]
Looks like the nightly is complete: `Version 3.11.99 (Jan 15 2019)`. I ran it just now and it works as expected. In other words, previous functionality is restored where only highlighted text is run. Thanks, again! :)

**Comment by justinclift** [comment]
Just in case you are, a Win64 build should appear here in the next ~20 minutes, assuming nothing went wrong:

* https://nightlies.sqlitebrowser.org/win64-onceoffs/DB.Browser.for.SQLite-issue1708v1-win64.msi

**Comment by justinclift** [comment]
Excellent @chrisjlocke.  Sounds like we have a winner! :smile:

Good stuff @MKleusberg + @beRto-.  That's one less bug for the release!

**Comment by chrisjlocke** [comment]
One quirk I found though, was related to the 'Execute current line'.  Guess it warrants its own issue rather than hanging around this one...

sql: 
```
1     select * from table1
2     where id=5;
3
4     select * from table1
5     where id=4;
6 
```
If lines 1 or 4 are highlighted, then the whole table is returned, as expected.
If the cursor is on lines 1, 4 or 5 and  'Execute current line' is pressed, then the relevant query (either just the first or second one) is run.  However.  If the cursor is on line 2 and 'Execute current line' is pressed, an error is returned:

    Execution finished with errors.
    -- At line 2:
    where
    -- Result: near "where": syntax error

I assume it shouldn't work on line 5 (it should report the same error as on line 2)
Additionally, if the cursor is on line 6 and 'Execute current line' is pressed, the log window is prefixed with 'Execution finished without errors." - it does not clear, nor do the results clear.  Keep pressing <kbd>Shift</kbd>+<kbd>F5</kbd> and the log window just fills up (from the top...)
